### PR TITLE
[SYTO-928] add CSS for the Broadsheet

### PR DESCRIPTION
**Pull request by asabaylus**
This change set introduces the layout and styles for the Broadsheet.
See: https://jira.natgeo.com/browse/SYTO-928

```
$ git checkout SYTO-928
$ gulp serve
```
Navigate to: http://127.0.0.1.xip.io:3000/components/#broadsheet

Verify that it all looks lovely!

PLEASE SQAUSH BEFORE MERGING! seriously I wrote a lot of this very late at night, and the commit history is total junk.

Note this is adaptive NOT responsive. We should probably make it responsive though cause while it meets MVP definition the fact you have to refresh to view the mobile correctly makes me nuts

### CODE REVIEW

File: src/modules/broadsheet/MTBroadsheet.js [hunk 1]
```diff
@@ -1,78 +1,234 @@
...
+    super(props);
+    this.debouncedReveal = null;
+    this.reveal = this.reveal.bind(this);
+    this.state = {
+      breakpoint: this.props.parentWidth || null
+    }
+  }
+
+  reveal() {
+    let el = this.refs.broadsheet;
```
Comment (pmourelle) [inline src/modules/broadsheet/MTBroadsheet.js]: One `let` to rule them all..

File: src/modules/broadsheet/MTBroadsheet.js [hunk 2]
```diff
@@ -1,78 +1,234 @@
...
+      window.removeEventListener('scroll', this.debouncedReveal);
+    };
+  }
+
+  componentDidMount() {
+    // wait a few seconds because the user may need download the contents/images
+    // before we start the animation if the broadsheet is inframe on initial
+    // page load
+    _delay(this.reveal, 2000);
+    this.debouncedReveal = _debounce(this.reveal, 500);
```
Comment (pmourelle) [inline src/modules/broadsheet/MTBroadsheet.js]: I'd put this line in the constructor.

File: src/modules/broadsheet/MTBroadsheet.js [hunk 3]
```diff
@@ -1,78 +1,234 @@
...
+  render() {
+    if (!this.props.caption && this.props.credit) {
+      return (<span/>);
+    }
+
+    return (
+      <figure className="mt3_broadsheet-leadMedia-attributions">
+        <figcaption className="mt3_caption-container--indent">
+          <p className="mt3_caption-body">
+            <span className="mt3_caption-title mt3_visuallyhidden">Photo Title{" "}</span>
```
Comment (pmourelle) [inline src/modules/broadsheet/MTBroadsheet.js]: `{" "}` ?

File: src/modules/broadsheet/_styles.scss [hunk 4]
```diff
@@ -0,0 +1,302 @@
+$mt3_small: 768px;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: You may use/extend `mortar/src/styles/themes/default/vars/_mqs.scss` maybe?

File: src/modules/broadsheet/_styles.scss [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+.mt3_broadsheet_wrapper {
+}
+.boo {
+    background: red;
+}
+
+// Layout
+.mt3_broadsheet_row {
+  display: -webkit-flex;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: Almost sure you don't need this.

File: src/modules/broadsheet/_styles.scss [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+  display: flex;
+  -webkit-flex-direction: row;
+  flex-direction: row;
+  align-items: stretch;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: AZ properties!

File: src/modules/broadsheet/_styles.scss [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+    line-height: 1.15;
+    text-transform: none;
+    margin: 0px 20% 30px;
+    @media(max-width: 375px) {
+      margin: 0px 10% 30px;
+    }
+    padding: 0;
+    display: block;
+    h1 {
+      font-size: 1.8rem;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: `rem` units is somthing to discuss as root's font size may vary depending on host site.

File: src/modules/broadsheet/_styles.scss [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+      clear: both;
+      float: left;
+      width: 100%;
+      height: 30px;
+    }
+  }
+
+  .mt3_broadsheet-story-also-in {
+    padding: 12px;
+    font-size: 0.7rem;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: Fonts and colors definitions. We have plenty in mortar, use them! :)

File: src/modules/broadsheet/_styles.scss [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+  }
+
+  .mt3_broadsheet-subscribe-label {
+    font-weight: 600;
+    letter-spacing: 0.1em;
+    line-height: 1.6;
+  }
+
+  section {
+      position: relative;
```
Comment (pmourelle) [inline src/modules/broadsheet/_styles.scss]: 2-4 spaces indentation mixed.

File: src/modules/broadsheet/MTBroadsheet.js (hunk 2 above)
Comment (asabaylus) [inline src/modules/broadsheet/MTBroadsheet.js]: Why? The `componentDidMount` react lifecycle dictate this is the point in time when the component has rendered into the DOM. It seemed like the best time to init the animation. On the constructor the animation may fire before the component is mounted.

File: src/modules/broadsheet/MTBroadsheet.js (hunk 3 above)
Comment (asabaylus) [inline src/modules/broadsheet/MTBroadsheet.js]: Forces an empty space. `&nbsp;` for  JSX will break the codes

File: src/modules/broadsheet/_styles.scss (hunk 5 above)
Comment (asabaylus) [inline src/modules/broadsheet/_styles.scss]: Almost isn't 100% ;)

File: src/modules/broadsheet/_styles.scss (hunk 7 above)
Comment (asabaylus) [inline src/modules/broadsheet/_styles.scss]: I did that per @hkekeli's remarks

File: src/modules/broadsheet/MTBroadsheet.js (hunk 2 above)
Comment (pmourelle) [inline src/modules/broadsheet/MTBroadsheet.js]: Right, but debounce() returns the function that you'll be calling later.

**Comment by asabaylus** [comment]
Please please fix stuff here! You can contribute to this PR. I'm all booked up and may not be able to get more work done on this ticket.

There's a couple of known issues. Brownie points for fixing them!

File: package.json [hunk 10]
```diff
@@ -99,6 +99,7 @@
     "metalsmith-markdown": "^0.2.1",
     "mocha": "^2.4.5",
     "phantomjs-prebuilt": "^2.1.7",
+    "prismjs": "^1.5.1",
```
Comment (asabaylus) [inline package.json]: @hkekeli leaving out this dep causes the npm i plus build to fail. Why am i the only one who had this issue?

File: src/modules/broadsheet/MTBroadsheet.js (hunk 1 above)
Comment (asabaylus) [inline src/modules/broadsheet/MTBroadsheet.js]: I dont get it? are you saying to use comma separated?

```
let x = 0, 
     y = 1,
      ...
```

???

File: src/modules/broadsheet/MTBroadsheet.js [hunk 11]
```diff
@@ -1,78 +1,234 @@
...
+// TODO: add shape validation for leadMedia and coverImage
 Broadsheet.propTypes = {
-  coverImage: PropTypes.string,
+  coverImage: PropTypes.object,
   issueDate: PropTypes.string,
   issueUrl: PropTypes.string,
-  leadMediaCaption: PropTypes.string,
-  leadMediaCredit: PropTypes.string,
-  leadMediaImage: PropTypes.string,
+  leadMedia: PropTypes.object,
```
Comment (asabaylus) [inline src/modules/broadsheet/MTBroadsheet.js]: needs _shape_ validation

File: src/modules/broadsheet/MTBroadsheet.js [hunk 12]
```diff
@@ -1,78 +1,234 @@
...
+        </div>
       </div>
     );
   }
 }
 
+// TODO: add shape validation for leadMedia and coverImage
 Broadsheet.propTypes = {
-  coverImage: PropTypes.string,
+  coverImage: PropTypes.object,
```
Comment (asabaylus) [inline src/modules/broadsheet/MTBroadsheet.js]: needs _shape_ validation

**Comment by benjachapman** [comment]
@Asa seems like there's a decent amount of things remaining that will need doing before this is usable in AEM - this doesn't seem to anticipate the DOM structure we'll need to enable the Smartbody to work, for example. I'd like product to prioritize this work before one of us jumps in. Thanks :)

**Comment by pmourelle** [comment]
@asabaylus @benjachapman I'm about to put release/3.0 into master, so this PR will have to be done against master and changes needed can be work in the meantime.

**Comment by pmourelle** [comment]
Re-created PR against `master` -> https://github.com/natgeo/mortar/pull/387

**Pull request closed without merging**